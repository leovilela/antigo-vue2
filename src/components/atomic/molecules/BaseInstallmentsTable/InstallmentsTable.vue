<template>
  <table class="table table-striped base-installment-table">
    <thead>
      <th colspan="2">&nbsp;</th>
      <th
        class="base-installment-table__title"
        colspan="2"
        v-for="(column, index) in columns"
        :key="index"
      >
        <strong>{{ column.label }}</strong>
      </th>
    </thead>
    <tbody>
      <tr
        v-if="isUniqRow"
        :class="{
          'base-installment-table__line': hasHighlightInstallment,
          'base-installment-table__line--active': ativo == -1
        }"
      >
        <td class="base-installment-table__installment-mode" colspan="2">
          <Base-Radio
            :name="installmentModeName"
            isInline
            :nativeValue="uniqRow"
            @input="onSelectedInstallment"
          >
            Parcela Ãºnica
          </Base-Radio>
        </td>
        <td
          class="base-installment-table__item"
          colspan="2"
          v-for="(column, index) in columns"
          :key="index"
        >
          {{ uniqRow[column.field] }}
        </td>
      </tr>
      <tr
        v-for="(row, index) in rows"
        :key="index"
        v-else
        :class="{
          'base-installment-table__line': hasHighlightInstallment,
          'base-installment-table__line--active': ativo == index
        }"
      >
        <td colspan="2" class="base-installment-table__installment-mode">
          <Base-Radio
            :name="installmentModeName"
            isInline
            :nativeValue="row"
            @input="onSelectedInstallment"
          >
            {{ row.installmentMode }}x
          </Base-Radio>
        </td>
        <td
          class="base-installment-table__item"
          colspan="2"
          v-for="(column, index) in columns"
          :key="index"
        >
          {{ row[column.field] }}
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script>
export default {
  name: "LInstallmentTable",
  props: {
    isUniqRow: {
      type: Boolean,
      default: false
    },
    hasHighlightInstallment: {
      type: Boolean,
      default: false
    },
    installmentModePrefix: {
      type: String,
      default: ""
    },
    columns: {
      type: Array,
      required: true
    },
    rows: {
      type: Array,
      required: true
    }
  },
  computed: {
    uniqRow() {
      if (!this.isUniqRow) {
        return {};
      }
      return this.rows[0];
    },
    installmentModeName() {
      const modeName = "installment-mode";
      return this.installmentModePrefix
        ? modeName + `-${this.installmentModePrefix}`
        : modeName;
    }
  },
  data() {
    return {
      ativo: -1
    };
  },
  methods: {
    onSelectedInstallment(installment) {
      this.ativo = installment.meta.parcelaOriginal;
      this.$emit("onSelectedInstallment", installment);
    }
  }
};
</script>

<style lang="less" scoped>
@import "./style.less";
</style>
