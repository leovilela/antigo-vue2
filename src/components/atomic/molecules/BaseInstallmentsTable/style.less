.base-installment-table {
  margin: 0;

  tr {
    &:hover{
      background: yellow;
    }
  }

  &__title {
    text-align: right;
    &:last-of-type {
      padding-right: 16px;
    }
  }
  &__item {
    text-align: right;
    height: 23px;
    padding: 0;
    vertical-align: middle;

    &:last-of-type {
      padding-right: 16px;
    }
  }
  &__installment-mode {
    text-align: left;
    height: 23px;
    padding: 0;
    padding-left: 16px;
  }
}
