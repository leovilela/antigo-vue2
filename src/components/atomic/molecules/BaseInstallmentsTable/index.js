import InstallmentsTable from "./InstallmentsTable";

export default InstallmentsTable;
