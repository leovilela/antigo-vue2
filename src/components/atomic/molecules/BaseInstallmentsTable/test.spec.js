import { shallowMount } from "@vue/test-utils"
import BaseInstallmentsTable from "./index"
let wrapper

describe("Component BaseInstallmentsTable <base-installment-table />", () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseInstallmentsTable)
  })

  afterEach(() => {
    wrapper.destroy();
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('LInstallmentTable')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

})
