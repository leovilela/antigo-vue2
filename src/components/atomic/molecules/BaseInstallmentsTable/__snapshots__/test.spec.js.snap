// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BaseInstallmentsTable <base-installment-table /> render correctly 1`] = `<div></div>`;
